Rakefile

```diff
@@ -262,6 +263,399 @@ task :xcode => [:dependencies] do
   sh "open #{XCODE_WORKSPACE}"
 end
 
+desc "Install and configure WordPress iOS and it's dependencies - External Contributors"
+namespace :init do
+task :oss => %w[
+  install:xcode:check
+  dependencies
+  install:tools:check_oss
+  install:lint:check
+  credentials:setup
+]
+
+desc "Install and configure WordPress iOS and it's dependencies - a8c Developers"
+task :developer => %w[
+  install:xcode:check
+  dependencies
+  install:tools:check_developer
+  install:lint:check
+  credentials:setup
+  gpg_key:setup
+]
+end
+
+namespace :install do
+  namespace :xcode do
+    task :check => %w[xcode_app:check xcode_select:check]
+
+    #xcode_app namespace checks for the existance of xcode on developer's machine,
+    #checks to make sure that developer is using the correct version per the CI specs
+    #and confirms developer has xcode-select command line tools, if not installs them
+    namespace :xcode_app do
+      #check the existance of xcode, and compare version to CI specs
+      task :check do
+        puts "Checking for system for XCode"
+        if !xcode_installed?
+          #if xcode is not installed, prompt user to install and terminate rake
+          puts "XCode not Found!"
+          puts ""
+          puts "====================================================================================="
+          puts "Developing for WordPressiOS requires XCode."
+          puts "Please install XCode before setting up WordPressiOS"
+          puts "https://apps.apple.com/app/xcode/id497799835?mt=12"
+          abort("")
+        else
+          puts "Xcode installed"
+        end
+
+        puts "Checking CI recommendded installed XCode version"
+
+        if !xcode_version_is_correct?
+          #if xcode is the wrong version, prompt user to install the correct version and terminate rake
+          puts "Not recommended version of XCode installed"
+          puts "It is recommended to use Xcode version #{get_ci_xcode_version}"
+          STDOUT.puts "Please press enter to continue"
+          complete = STDIN.gets.strip
+          next
+        end
+      end
+
+      #Check if XCode is installed
+      def xcode_installed?
+        system "xcodebuild -version", [:out, :err] => File::NULL
+      end
+
+      #compare xcode version to expected CI spec version
+      def xcode_version_is_correct?
+        if get_xcode_version == get_ci_xcode_version
+          puts "Correct version of XCode installed"
+          return true
+        end
+      end
```

Comment:
Is it ok for this to return `nil` if the two versions don't match? Or would it be clearer if it returned false?

<img width="537" alt="Screen Shot 2020-05-27 at 3 22 40 pm" src="https://user-images.githubusercontent.com/1218433/82980922-07525f00-a02e-11ea-8719-f7397b475c7a.png">

Reply:
That's a good question.  I had not added any return to false because what it triggers happen if what comes back is not true.  So my thought on this was to only create the circumstance t return true, and left out other cases.  
But since it is a boolean check it would be clearer to return false if  the check fails.